checks/prometheus_check.py

```diff
@@ -120,25 +124,54 @@ def parse_metric_family(self, buf, content_type):
                 yield message
         elif 'text/plain' in content_type:
             messages = {}  # map with the name of the element (before the labels) and the list of occurrences with labels and values
-            obj_map = {}   # map of the types of each metrics
+            obj_map = {}  # map of the types of each metrics
             obj_help = {}  # help for the metrics
-            for line in buf.splitlines():
-                self._extract_metrics_from_string(line, messages, obj_map, obj_help)
-
-            # Add type overrides:
-            for m_name, m_type in self.type_overrides.iteritems():
-                if m_type in self.METRIC_TYPES:
-                    obj_map[m_name] = m_type
-                else:
-                    self.log.debug("type override %s for %s is not a valid type name" % (m_type,m_name))
+            for metric in text_string_to_metric_families(buf):
```

Comment:
Instead of loading the whole response body in memory, we should stream the body, use [iter_lines](http://docs.python-requests.org/en/master/api/#requests.Response.iter_lines) as a generator to be fed into [text_fd_to_metric_families](https://github.com/prometheus/client_python/blob/master/prometheus_client/parser.py#L146)